R/utils-numeric.R

```diff
@@ -1,3 +1,45 @@
+## TODO think about the relationship between this and the plot.data bin fxns
+#' Find Bin Ranges
+#' 
+#' This function will find bin start, end and labels for a 
+#' continuous variable. Optionally, it can return a value/ count
+#' per bin. By default returns 10 bins for equalRanges and quantile
+#' methods and 6 for sd (standard deviations).
+#' @param x Numeric (or Date) vector to find bins for
+#' @param method A string indicating which method to use to find bins ('equalRanges', 'quantile', 'sd')
+#' @param numBins A number indicating how many bins are desired
+#' @param getValue A boolean indicating whether to return the counts per bin
+#' @export 
+getBinRanges <- function(x, method = c('equalInterval', 'quantile', 'sd'), numBins = NULL, getValue = c(TRUE, FALSE)) {
+  #validate inputs, if numBins is NULL set it
+  #if method is sd, ignore numBins?
+
+  binEdges <- unname(breaks(x, method, numBins))
+  if (method == 'quantile' && anyDuplicated(binEdges)) {
+    ## TODO decide if we prefer this, or an error, for UX
+    warning("There is insufficient data to produce the requested number of bins. Returning as many bins as possible.")
+    binEdges <- unique(binEdges)
```

Comment:
Non-unique SD breaks are handled in the `breaks()` function, while quantile breaks are handled here. Is that OK?

Reply:
I did that only bc we are using breaks() other places and didn't want to change the behavior. I could make them consistent, but I'd have to put some thought to it

Reply:
all should be consistent now. i think in an ideal world the breaks fxn would handle this, but i dont think its worth updating now. so ill just handle dups here regardless of method i guess